test/boost/sstable_compaction_test.cc

```diff
@@ -444,12 +444,13 @@ static future<std::vector<unsigned long>> compact_sstables(test_env& env, sstrin
             return compact_sstables(cf.get_compaction_manager(), sstables::compaction_descriptor(std::move(sstables_to_compact),
                 default_priority_class()), *cf, new_sstable).then([generation] (auto) {});
         } else if (strategy == compaction_strategy_type::leveled) {
+            std::vector<sstables::shared_sstable> candidates;
+            candidates.reserve(sstables->size());
             for (auto& sst : *sstables) {
                 BOOST_REQUIRE(sst->get_sstable_level() == 0);
                 BOOST_REQUIRE(sst->data_size() >= min_sstable_size);
-                column_family_test(cf).add_sstable(sst);
+                candidates.push_back(sst);
             }
-            auto candidates = get_candidates_for_leveled_strategy(*cf);
```

Comment:
let's break this into a patch of its own.
and justify it there.

Reply:
That change was required as part of the change, as that unit test doesn't run in a thread, and get_candidates_for_leveled_strategy() would fail into is_running_in_a_thread assertion.

Reply:
I can move this change into a separate patch that comes before the change in column_family_test.

Reply:
thanks